api/watcher/watcher.go

```diff
@@ -777,35 +777,17 @@ type machineAttachmentsWatcher struct {
 	out                         chan []watcher.MachineStorageID
 }
 
-// NewVolumeAttachmentsWatcher returns a MachineStorageIdsWatcher which
-// communicates with the VolumeAttachmentsWatcher API facade to watch
-// volume attachments.
-func NewVolumeAttachmentsWatcher(caller base.APICaller, result params.MachineStorageIdsWatchResult) watcher.MachineStorageIDsWatcher {
-	return newMachineStorageIdsWatcher("VolumeAttachmentsWatcher", caller, result)
-}
-
-// NewVolumeAttachmentPlansWatcher returns a MachineStorageIdsWatcher which
-// communicates with the VolumeAttachmentPlansWatcher API facade to watch
-// volume attachments.
-func NewVolumeAttachmentPlansWatcher(caller base.APICaller, result params.MachineStorageIdsWatchResult) watcher.MachineStorageIDsWatcher {
-	return newMachineStorageIdsWatcher("VolumeAttachmentPlansWatcher", caller, result)
-}
-
-// NewFilesystemAttachmentsWatcher returns a MachineStorageIdsWatcher which
-// communicates with the FilesystemAttachmentsWatcher API facade to watch
-// filesystem attachments.
-func NewFilesystemAttachmentsWatcher(caller base.APICaller, result params.MachineStorageIdsWatchResult) watcher.MachineStorageIDsWatcher {
-	return newMachineStorageIdsWatcher("FilesystemAttachmentsWatcher", caller, result)
-}
-
-func newMachineStorageIdsWatcher(facade string, caller base.APICaller, result params.MachineStorageIdsWatchResult) watcher.MachineStorageIDsWatcher {
+// NewStorageAttachmentsWatcher returns a MachineStorageIdsWatcher which
+// communicates with the StorageAttachmentsWatcher API facade to watch
+// storage attachments.
+func NewStorageAttachmentsWatcher(caller base.APICaller, result params.MachineStorageIdsWatchResult) watcher.MachineStorageIDsWatcher {
 	w := &machineAttachmentsWatcher{
 		caller:                      caller,
 		machineAttachmentsWatcherId: result.MachineStorageIdsWatcherId,
 		out:                         make(chan []watcher.MachineStorageID),
 	}
 	w.tomb.Go(func() error {
-		return w.loop(facade, result.Changes)
+		return w.loop("StorageAttachmentsWatcher", result.Changes)
```

Comment:
This isn't correct is it? This is the facade that needs to be called why is it now hardcoded?

Reply:
For backwards compatibility, we decided to keep the 3 existing attachment facades.